srml/support/src/dispatch.rs

```diff
@@ -811,9 +811,21 @@ macro_rules! __impl_outer_dispatch_metadata {
 			pub fn outer_dispatch_metadata() -> $crate::dispatch::OuterDispatchMetadata {
 				$crate::dispatch::OuterDispatchMetadata {
 					name: $crate::dispatch::DecodeDifferent::Encode(stringify!($outer_name)),
-					calls: __impl_outer_dispatch_metadata!(@encode_calls 0; ; $( $module::$call, )*),
+					calls: $crate::dispatch::DecodeDifferent::Encode(
+						__impl_outer_dispatch_metadata!(@encode_calls 0; ; $( $module::$call, )*)
+					),
 				}
 			}
+			pub fn module_dispatch(mod_name: &'static str) -> Option<$crate::dispatch::OuterDispatchCall> {
+				let metas = __impl_outer_dispatch_metadata!(@encode_calls 0; ; $( $module::$call, )*);
+				for i in metas.iter() {
+					if i.prefix == $crate::dispatch::DecodeDifferent::Encode(mod_name) {
```

Comment:
No please, don't do that. You are encoding the same value over and over. This will kill the performance.

I would accept a solution that uses https://crates.io/crates/paste to create custom functions per outer dispatch.